src/decoders/http2/http2_decoder.c

```diff
@@ -246,7 +251,10 @@ void qd_http2_decoder_connection_free(qd_http2_decoder_connection_t *conn_state)
  */
 static int decompress_headers(qd_http2_decoder_t *decoder, uint32_t stream_id, const uint8_t *data, size_t length)
 {
-    assert(length > 0);
+    if (length <= 0) {
+        qd_log(LOG_HTTP2_DECODER, QD_LOG_ERROR, "[C%"PRIu64"] decompress_headers - passed in length < 0", decoder->conn_state->conn_id);
```

Comment:
I don't think this should be logged at the ERROR level, maybe DEBUG or not at all.  We don't want an external process to be able to clog up our logs by sending garbage protocol traffic.